### PR TITLE
Automate generation of codemeta.json and .zenodo.json files

### PR DESCRIPTION
**Pull request by Bultako**
This PR intends to solve issue #3775 

It uses [cffconvert-github-action](https://github.com/citation-file-format/cffconvert-github-action) and [eossr tools](https://escape2020.pages.in2p3.fr/wp3/eossr/index.html) so to create `codemeta.json` and `.zenodo.json` software metadata description and index files from a `CITATION.cff` file. The action is triggered for every modification of `CITATION.cff` file and also validates the content of this file.

The workflow is:
`CITATION.cff -> cffconvert -> fix codemeta.json -> eossr validator -> eossr converter -> .zenodo.json`

### CODE REVIEW

**Comment by Bultako** [comment]
I would suggest to review and merge this PR, and then create another for the [on-boarding of gammapy in ESCAPE via Zenodo](https://github.com/gammapy/gammapy/issues/3847) once we have the answers to some of the questions exposed there.

File: CITATION.cff [hunk 1]
```diff
@@ -15,7 +15,6 @@ authors:
   affiliation: Center for Astrophysics | Harvard & Smithsonian, CfA, 60 Garden St.,
     02138 Cambridge MA, US
   orcid: https://orcid.org/0000-0003-4568-7005
-  email: "axel.donath@cfa.havard.edu"
```
Comment (adonath) [inline CITATION.cff:18]: Why does it remove the mail address here?

Comment (adonath) [reply CITATION.cff:18]: Isn't it support anymore in cff v1.2.0?

Comment (Bultako) [reply CITATION.cff]: You're right !
The email attribute is allowed in the cff v1.2.0 schema.
I have recovered it in last commit.

**Comment by Bultako** [comment]
> Can cffconvert just append/update an existing codemeta.json file ?

I don't think so.

**Comment by vuillaut** [comment]
See comment https://github.com/gammapy/gammapy/issues/3847#issuecomment-1069117448  related to this PR.

**Review by adonath — APPROVED** [review]
Thanks a lot @Bultako, no further comments from my side. Maybe @vuillaut can do the final review?

**Comment by vuillaut** [comment]
> Thanks a lot @Bultako, no further comments from my side. Maybe @vuillaut can do the final review?

Yes, I will have a look early next week, I think there are still some points to clear.

**Comment by vuillaut** [comment]
Ok, some comments:
- the version in `.zenodo.json` must be a string
- the complete list of communities (and in general of metadata) must be included in the `.zenodo.json` or former ones will be removed [(I will include that in our doc)](https://gitlab.in2p3.fr/escape2020/wp3/eossr/-/issues/88)

The complete Zenodo metadata for gammapy v0.19 can be found [here](https://zenodo.org/record/5721467/export/json#.YjrXyTdKj0o), this can help you create a more complete one in the CI.
However, if you changed some metadata manually in Zenodo web interface (such as adding some communities) after the release of v0.19, they won't appear there and you must retrieve them and add  them to the `.zenodo.json`.

[I intend to create a GitHub action to validate a `.zenodo.json`](https://gitlab.in2p3.fr/escape2020/wp3/eossr/-/issues/89) that you could use in the CI. In the meantime you can also use the already existing CLI `eossr-check-connection-zenodo`, [here is how](https://escape2020.pages.in2p3.fr/wp3/eossr/snippets/4.ex_CI_test_codemeta_master.html#id1).

As always, your feedback is welcome and you let me know if I can help.

**Review by vuillaut — CHANGES_REQUESTED** [review]
Some changes must be addressed or the upload to Zenodo will fail.

**Comment by Bultako** [comment]
Actually, I have just found that zenodo supports `CITATION.cff` files, in the sense that the list of contributors considered in the zenodo record is extracted from the `CITATION.cff` file present at the moment of making a release in the GitHub repo. So, I do not see the need of having a `.zenodo.json` file anymore. Moreover, it seems having a `.zenodo.json` is useless when a `CITATION.cff` file is present. 

See https://github.com/citation-file-format/citation-file-format/issues/374#issuecomment-1061022832

This simplifies much more the things since we get rid of synchronizing zenodo metadata with a `.zenodo.json` file that potentially removes other internal zenodo metadata.

This PR would then be also more simple, it would trigger when modifs in the content of `CITATION.cff` file only updating the `codemeta.json` file needed by the ESCAPE 2020 folks. If there is some fine-tuning to do in the zenodo metadata record, this can be done using their online record edition tool.

If you are ok with this I can simplify this GitHub action removing the step that produces the `.zenodo.json` file.

**Comment by adonath** [comment]
What is the status here @Bultako, anything left to be done?

**Comment by Bultako** [comment]
@adonath @vuillaut 
from my side this PR is ready to merge.

**Review by adonath — DISMISSED** [review]
Thanks a lot @Bultako, I think it is a very nice a clean solution now!

File: dev/codemeta.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import json
+
+FILENAME = "../codemeta.json"
+
+# add potentially missing content
+with open(FILENAME, "r") as f:
+    data = json.load(f)
+data["maintainer"] = data["author"][0]
+data["readme"] = "https://gammapy.org"
+data["issuetracker"] = "https://github.com/gammapy/gammapy/issues"
```
Comment (vuillaut) [inline dev/codemeta.py:11]: `issueTracker` instead of `issuetracker`

**Review by vuillaut — CHANGES_REQUESTED** [review]

File: codemeta.json [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+                "name": "Max Planck Institut f\u00fcr Kernphysik, MPIK, Saupfercheckweg 1, 69117 Heidelberg, Germany"
+            },
+            "familyName": "Olivera",
+            "givenName": "Laura"
+        },
+        {
+            "@type": "Person",
+            "affiliation": {
+                "@type": "Organization",
+                "name": "Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"
```
Comment (bkhelifi) [inline codemeta.json:114]: the name of his affiliation is:
"Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris Cit\u00e9, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"

File: codemeta.json [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+            },
+            "familyName": "Deil",
+            "givenName": "Christoph"
+        },
+        {
+            "@id": "https://orcid.org/0000-0002-8219-4667",
+            "@type": "Person",
+            "affiliation": {
+                "@type": "Organization",
+                "name": "Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"
```
Comment (bkhelifi) [inline codemeta.json:29]: the name of his affiliation is:
"Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris Cit\u00e9, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"

File: codemeta.json [hunk 5]
```diff
@@ -0,0 +1,597 @@
...
+                "name": "Friedrich-Alexander-Universit\u00e4t Erlangen-N\u00fcrnberg, Erlangen Centre for Astroparticle Physics, Erwin-Rommel-Str. 1 91058 Erlangen, Germany"
+            },
+            "familyName": "Nakashima",
+            "givenName": "Kaori"
+        },
+        {
+            "@type": "Person",
+            "affiliation": {
+                "@type": "Organization",
+                "name": "Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"
```
Comment (bkhelifi) [inline codemeta.json:412]: the name of her affiliation is:
"Astroparticle and Cosmology Laboratory CNRS, Universit\u00e9 de Paris Cit\u00e9, 10 Rue Alice Domon et L\u00e9onie Duquet, 75013 Paris, France"

File: codemeta.json [hunk 6]
```diff
@@ -0,0 +1,597 @@
...
+            },
+            "familyName": "Giunti",
+            "givenName": "Luca"
+        },
+        {
+            "@type": "Person",
+            "affiliation": {
+                "@type": "Organization",
+                "name": "CNRS"
```
[the remark this answers is not in the document]
Comment (adonath) [reply codemeta.json:123]: @bkhelifi Ideally you open a separate PR and fix the affiliation directly in https://github.com/gammapy/gammapy/blob/master/CITATION.cff#L65 .  The file you commented is generated automatically from `CITATION.cff` now.

File: codemeta.json (hunk 3 above)
Comment (vuillaut) [reply codemeta.json:114]: I believe this should be changed in the `CITATION.cff` file (and probably in another PR)

File: dev/codemeta.py (hunk 2 above)
Comment (Bultako) [reply dev/codemeta.py]: ok, done

**Review by vuillaut — APPROVED** [review]
The `issueTracker` key is still `issuetracker` in the `codemeta.json` file but that will be changed when actions run.
Great PR, thank you very much for the productive exchanges @Bultako .

**Comment by adonath** [comment]
Thanks @Bultako for the work and @vuillaut for the review! I'll go ahead and merge this, the `codemeta.json` will then be updated when introducing the next changes to `CITATION.cff`...

**Review by adonath — APPROVED** [review]

**Comment by adonath** [comment]
@bkhelifi Please open a new PR changing your affiliation in `CITATION.cff`, everything else will then be updated automatically...

**Pull request merged**